### PR TITLE
Test folder with special characters in name

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Add acceptance test to create a folder with special characters in its name and then browse to that folder 

## Related Issue

- Related to #2467

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6572/

<p>





</p></details>

**Comment by jasson99** [comment]
@skshetry

File: tests/acceptance/features/webUIFiles/createFolders.feature [hunk 1]
```diff
@@ -38,6 +38,15 @@ Feature: create folders
     When the user creates a folder with the invalid name "folder/subfolder" using the webUI
     Then the error message 'Folder name cannot contain "/"' should be displayed on the webUI dialog prompt
 
+  @issue-2467 @yetToImplement
+  Scenario: Create folder with special characters in its name and browse to that folder
+    When the user creates a folder with the name "?&%0" using the webUI
+    And the user opens folder "?&%0" using the webUI
+    And the user creates a folder with the name "folder-inside-folder" using the webUI
```
Comment (skshetry) [inline tests/acceptance/features/webUIFiles/createFolders.feature]:
```suggestion
```

File: tests/acceptance/features/webUIFiles/createFolders.feature [hunk 2]
```diff
@@ -38,6 +38,15 @@ Feature: create folders
...
+    And the user reloads the current page of the webUI
+    Then the files list should not be present on the webUI
```
Comment (skshetry) [inline tests/acceptance/features/webUIFiles/createFolders.feature]:
```suggestion
    Then the files list should be empty
```

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 3]
```diff
@@ -539,6 +539,12 @@ module.exports = {
         .click(formattedFileRow)
         .waitForElementVisible('@versionsElement')
         .click('@versionsElement')
+    },
+    assertElementNotPresent: function () {
```
Comment (skshetry) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: Implement this function in `phoenixPage` and check for element not present `#Phoenix`.
```suggestion
    assertPageIsEmpty: function () {
```

**Comment by jasson99** [comment]
@skshetry @individual-it @dpakach

File: tests/acceptance/features/webUIFiles/createFolders.feature [hunk 4]
```diff
@@ -38,6 +38,14 @@ Feature: create folders
     When the user creates a folder with the invalid name "folder/subfolder" using the webUI
     Then the error message 'Folder name cannot contain "/"' should be displayed on the webUI dialog prompt
 
+  @issue-2467 @yetToImplement
```
Comment (dpakach) [inline tests/acceptance/features/webUIFiles/createFolders.feature]:
```suggestion
  @issue-2467
```

File: tests/acceptance/pageObjects/phoenixPage.js [hunk 5]
```diff
@@ -116,6 +116,9 @@ module.exports = {
           .click(declineShareButton)
           .waitForAjaxCallsToStartAndFinish()
       }
+    },
+    assertPageEmpty: function () {
```
Comment (dpakach) [inline tests/acceptance/pageObjects/phoenixPage.js]: Please dont use assertions in page objects.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6695/

<p>



































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6695/

<p>







































































































</p></details>

File: tests/acceptance/pageObjects/phoenixPage.js [hunk 6]
```diff
@@ -116,6 +116,13 @@ module.exports = {
           .click(declineShareButton)
           .waitForAjaxCallsToStartAndFinish()
       }
+    },
+    getPhoenixcontainer: async function () {
+      let phoenixContainerContains = true
+      await this.api.elements('xpath', '@phoenixContainer', function (result) {
```
Comment (kiranparajuli589) [inline tests/acceptance/pageObjects/phoenixPage.js]: Dont use locateStrategy like this...
Fetch it from required elements object

Comment (jasson99) [reply tests/acceptance/pageObjects/phoenixPage.js]: done

File: tests/acceptance/pageObjects/phoenixPage.js [hunk 7]
```diff
@@ -116,6 +116,13 @@ module.exports = {
           .click(declineShareButton)
           .waitForAjaxCallsToStartAndFinish()
       }
+    },
+    getPhoenixcontainer: async function () {
```
Comment (skshetry) [inline tests/acceptance/pageObjects/phoenixPage.js]: Naming is wrong. It returns boolean, not the container.

File: tests/acceptance/pageObjects/phoenixPage.js [hunk 8]
```diff
@@ -116,6 +116,13 @@ module.exports = {
...
+      let phoenixContainerContains = true
```
Comment (skshetry) [inline tests/acceptance/pageObjects/phoenixPage.js]: Please rename the variable into something that is expressive and precise.

File: tests/acceptance/pageObjects/phoenixPage.js (hunk 7 above)
Comment (jasson99) [reply tests/acceptance/pageObjects/phoenixPage.js]: Would you mind recommending some?

File: tests/acceptance/pageObjects/phoenixPage.js [hunk 9]
```diff
@@ -116,6 +116,18 @@ module.exports = {
           .click(declineShareButton)
           .waitForAjaxCallsToStartAndFinish()
       }
+    },
+    /**
+     * Checks if the phoenix page contains any elements on the phoenix container
+     *
+     * @returns {boolean} Does the page contain any element on the phoenix container?
```
Comment (skshetry) [inline tests/acceptance/pageObjects/phoenixPage.js]:
```suggestion
     * @returns {boolean}
```

File: tests/acceptance/pageObjects/phoenixPage.js [hunk 10]
```diff
@@ -116,6 +116,18 @@ module.exports = {
...
       }
+    },
+    /**
+     * Checks if the phoenix page contains any elements on the phoenix container
+     *
+     * @returns {boolean}
+     */
+    isPageVisible: async function () {
+      let isVisible = true
+      await this.api.elements(this.elements.phoenixContainer.locateStrategy, this.elements.phoenixContainer.selector, function (result) {
```
Comment (skshetry) [inline tests/acceptance/pageObjects/phoenixPage.js]: Line too long. Otherwise, everything looks good.

Comment (skshetry) [inline tests/acceptance/pageObjects/phoenixPage.js]: Line too long. Otherwise, everything looks good.

**Pull request merged**